### ISSUE TITLE
ML: classificazione andamento con boosting

### ISSUE THREAD

**Issue by Tarekun** [Tarekun/zecca#31]
tecniche di boosting paiono essere tra le piu efficaci per dataset tabellari (ie i nostri prezzi+indicatori giornalieri). vale la pena fare un tentativo con questi approcci sul nostro dataset. bisogna individuare un insieme di classi da calcolare a partire dal prezzo, che dicano se il prezzo sale/scende. xgboost e' in assoluto il piu famoso e quello da cui partire, ma non e' l'unico disponibile (vedi catboost, adabost, lightgbm)

DA FARE
* implementazione di codice pandas che dato un dataframe di prezzi sia in grado di calcolare le label di classi riga per riga
* trovare una libreria python per boosting
* implementazione del training di un modello via boosting
* evaluation del modello

INCOGNITE
* quante classi? la base piu semplice sarebbe solo il giudizio binario sale/scende, ma e' poco espressivo. un compromesso soddisfacente potrebbe essere 5 classi: sale tanto, sale poco, praticamente non si muove, scende poco, scende tanto, ma non saprei dove mettere le soglie (su cosa poi? il prezzo stesso, il ritorno, ...). ideale sarebbe provare diversi insiemi di classi e vedere come cambiano i risultati
* che libreria? non ne ho idea, idealmente un'implementazione in una libreria con ampio supporto a ML/DS tipo scikitlearn

INDICAZIONI VALUTAZIONE
calcolare solo l'accuracy non è sufficiente: se il ~90% degli stock non "sale di tanto" nella finestra fissata, un modello puo ottenere accuracy del 90% predicendo sempre di no. presentare accuracy con precision e recall per classe, eventualmente f1 score. se il codice lo permette e non è troppo difficile fare cross validation non sarebbe male
